src/App.js

```diff
@@ -14,11 +14,86 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    basket: [],
+  }
+
+  toToggle = ({ target }) => {
+    const { basket } = this.state;
+    const titleOfGood = target.parentElement.firstChild.textContent;
+
+    target.parentElement.classList.toggle('selected');
+
+    basket.includes(titleOfGood)
+      ? this.toCancel(titleOfGood)
+      : this.toSelect(titleOfGood);
+  }
+
+  toSelect = (title) => {
+    const { basket } = this.state;
+
+    this.setState({ basket: [...basket, title] });
+  };
+
+  toCancel = (title) => {
+    const { basket } = this.state;
+    const index = basket.indexOf(title);
+    const itemsOfBasket = basket;
+
+    itemsOfBasket.splice(index, 1);
+    this.setState({ basket: [...itemsOfBasket] });
```

Comment:
if you want to remove an element from the array it's easier to use the filter method
but you don't need here array it should be only one selected item